packages/dapp/src/components/listing/CommitVoteDetail.tsx

```diff
@@ -0,0 +1,155 @@
+import * as React from "react";
+import styled from "styled-components";
+import TransactionButton from "../utility/TransactionButton";
+import { TwoStepEthTransaction } from "@joincivil/core";
+import { commitVote, requestVotingRights } from "../../apis/civilTCR";
+import BigNumber from "bignumber.js";
+
+const StyledFormContainer = styled.div`
+  margin: 1em 0;
+`;
+
+const FormValidationMessage = styled.div`
+  color: #c00;
+  font-weight: bold;
+`;
+
+const FormGroup = styled.div`
+  margin: 0 0 1em;
+`;
+
+export interface CommitVoteDetailProps {
+  challengeID: BigNumber;
+}
+
+export interface CommitVoteDetailState {
+  isVoteTokenAmtValid: boolean;
+  voteOption?: string;
+  salt?: string;
+  numTokens?: string;
+}
+
+export interface InputElementProps {
+  type: string;
+  name?: string;
+  value?: string;
+  readOnly?: boolean;
+  onChange?(event: any): void;
+  validate?(event: any): boolean;
+}
+
+class InputElement extends React.Component<InputElementProps> {
+  constructor(props: any) {
+    super(props);
+  }
+
+  public render(): JSX.Element {
+    return (
+      <input
+        type={this.props.type}
+        name={this.props.name}
+        value={this.props.value}
+        readOnly={this.props.readOnly}
+        onChange={this.onChange.bind(this)}
+      />
+    );
+  }
+
+  private onChange(event: any): void {
+    if (this.props.onChange) {
+      this.props.onChange(event);
+    }
+
+    if (this.props.validate) {
+      this.props.validate(event);
+    }
+  }
+}
+
+class CommitVoteDetail extends React.Component<CommitVoteDetailProps, CommitVoteDetailState> {
+  constructor(props: any) {
+    super(props);
+
+    this.state = {
+      isVoteTokenAmtValid: false,
+    };
+
+    this.updateCommitVoteParam = this.updateCommitVoteParam.bind(this);
+  }
+
+  public render(): JSX.Element {
+    return (
+      <StyledFormContainer>
+        <h3>Vote On Challenge</h3>
+
+        {/* @TODO/jon: Change this to a hidden form field.
```

Comment:
can you change "@TODO/jon" to "TODO(am9u):" ? easier to search for TODOs by person if we're all using the same format